pyigm/cgm/cgmsurvey.py

```diff
@@ -149,7 +156,133 @@ def nsys(self):
         -------
         nsys : int
         """
-        return len(self.cgm_abs)
+        if self.mask is not None:
+            return np.sum(self.mask)
+        elif len(self._data) > 0:
+            return len(self._data)
+        elif len(self._dict) > 0:
+            return len(self._dict)
+        else:
+            return len(self.cgm_abs)
+
+    def add_ion_to_data(self, inp_ion):
+        """ Add columns for ion info (column density)
+        to the Table
+
+        Parameters
+        ----------
+        inp_ion
+
+        Returns
+        -------
+
+        """
+        from linetools.analysis import absline as ltaa
+        ion = inp_ion.replace(' ','')
+        # Check for previous
+        if 'flag_N_{:s}'.format(ion) in self._data.keys():
+            print("Ion data is already in the _data table")
+            return
+        # Loop on the systems
+        flagNs, Ns, sigNs = [], [], []
+        for key in self._dict.keys():
+            igm_comp = self._dict[key]['igm_sys']['components']
+            comps = [] # Allow for more than one
+            for comp in igm_comp.keys():
+                sion = comp.split('_')[0]
+                if sion == ion:
+                    if 'attrib' in igm_comp[comp].keys():
+                        attrib = igm_comp[comp]['attrib']
+                        attrib['sig_logN'] = np.array(attrib['sig_logN'])
+                        comps.append(attrib.copy())
+                    else: # Deprecated
+                        comps.append(dict(logN=igm_comp[comp]['logN'],
+                                          flag_N=igm_comp[comp]['flag_N'],
+                                          sig_logN=np.array([igm_comp[comp]['sig_logN']]*2)))
```

Comment:
Is a check not necessary to see if igm_comp[comp]['sig_logN'] contains 2-sided limits already?